### ISSUE TITLE
JIT match failure for empty capture group

### ISSUE THREAD

**Issue by addisoncrump** [PCRE2Project/pcre2#336]
Discovered by #322.

This seems to be an outright incorrect behaviour in the JIT, and it's not an extreme edge case.

```
  re> /(1|)b*1/
data> (1|)b*1       
No match
data> (1|)b*1\=no_jit
 0: 1
 1: 
```

Note that the following cases operate as expected:
```
  re> /1?a*1/
data> 1
 0: 1
data> 
  re> /(1?)a*1/
data> 1
 0: 1
 1: 
data> 1\=no_jit
 0: 1
 1:
```

**Comment by carenas** [comment]
worth also that is not a problem in the legacy PCRE and it doesn't seem to be a regression from 10.42

```
% pcretest -s++
PCRE version 8.45 2021-06-15

  re> /(1|)b*1/
data> 1
 0: 1 (JIT)
 1:  (JIT)
```

**Comment by zherczeg** [comment]
Fixed in `2b926df83600f952a0d74ddcbcad1c840831fd76`
